app/code/Magento/Customer/view/frontend/templates/js/customer-data.phtml

```diff
@@ -5,21 +5,21 @@
  */
 
 /** @var \Magento\Customer\Block\CustomerData $block */
-
-// phpcs:disable Magento2.Templates.ThisInTemplate.FoundHelper
+/** @var \Magento\Customer\ViewModel\Customer\Data $viewModel */
+$viewModel = $block->getViewModel();
```

Comment:
Strictly speaking, this is backwards incompatible.
Example: when you've implement a new custom block in xml and referenced this phtml file. After these changes are added in core Magento, your custom block will start failing because there will be no `view_model` attribute set to it.

Maybe a similar approach should be taken here as when adding a new dependency to a class? And do something like this: `$viewModel = $block->getViewModel() ?? ObjectManager::getInstance()->get(\ Magento\Customer\ViewModel\Customer\Data::class);` ?

Question is: how many people will have added a new custom block that uses this phtml file, probably not a lot ...

Reply:
@hostep thank you for your review. If have a custom part involved then this solution will not work as expected. So I will think about another solution

Reply:
Hi @hostep, I added your suggest to new commit and added some change. Can you review again ?

Reply:
Thanks!
I'm leaving in holiday in a few days for 3 weeks, so let's hope somebody else will be able to fully review this PR in the meantime 🙂